README.md

```diff
@@ -45,4 +46,26 @@ Test your new function; I've provided some basic testing code in `code.test.js`.
 
 What is the worst-case asymptotic time complexity of your implementation? What
 is the worst-case asymptotic memory complexity? Add your answer, including your
-reasoning, to this markdown file.
+reasoning, to this markdown file.  
+
+**NOTE**:  
+Please note that I used the following code for reference in order to help me  
+understand how to iterate over all the paths in the list of cities:  
+https://github.com/COSC3020/tsp-held-karp-CadeMaynard/tree/main  
+In addition to this, I also used the slides provided in class in order to see how  
+memoization could best be done in this scenario.  
+
+**ANSWER**:  
+The worst case time complexity of this problem will depend almost entirely on the amount of nodes  
+that are included at the very beginning. This is because the code must go through every possible  
+combination of nodes in order to ensure that it's getting the best possible answer. Luckily, I've  
+already found the runtime analysis for such a scenario in the "Brute Force Sorting" algorithm that  
+I've done before. From that, I know that the worst case time complexity for trying out all possible  
+unique permutations of a set would be $\Theta(|V|!)$, so that would be our answer.
```

Comment:
Ok, in the worst case you would go through all possible cache values, right? But it's not just the cities that go into the cache, it's also the start city. And if you have the same start city and different permutations of the remaining cities, that gives you the same result, so should be in the same cache entry. Is that the case at the moment? If not, how could you make it so? And then what would the complexity be?